src/traversals/eulerian.jl

```diff
@@ -0,0 +1,103 @@
+# Adapted from SimpleGraphs.jl [Copyright (c) 2014, Ed Scheinerman]:
+#    https://github.com/scheinerman/SimpleGraphs.jl/blob/master/src/simple_euler.jl
+# Reproduced under the MIT Expat License.
+
+"""
+    eulerian(g::AbstractSimpleGraph{T}[, u::T]) --> T[]
+
+Returns a [Eulerian trail or cycle](https://en.wikipedia.org/wiki/Eulerian_path) through an
+undirected graph `g`, starting at vertex `u`, returning a vector listing the vertices of `g`
+in the order that they are traversed. If no such trail or cycle exists, throws an error.
```

Comment:
How do I know whether the result will be a trail or a cycle? What does it depend on?

Reply:
Only on the degree of the starting vertex?